test/new-e2e/tests/agent-platform/common/test_client.go

```diff
@@ -281,3 +282,93 @@ func ReadJournalCtl(t *testing.T, client *TestClient, grepPattern string) string
 	}
 	return journalCtlOutput
 }
+
+// DockerTestClient is a helper to run commands on a docker container for tests
+type DockerTestClient struct {
+	host          *components.RemoteHost
+	containerName string
+}
+
+// NewDockerTestClient creates a client to help write tests that run on a docker container
+func NewDockerTestClient(host *components.RemoteHost, containerName string) DockerTestClient {
+	return DockerTestClient{
+		host:          host,
+		containerName: containerName,
+	}
+}
+
+// RunContainer starts the docker container in the background based on the given image reference
+func (c *DockerTestClient) RunContainer(image string) error {
```

Comment:
Since this is a `framework` functionality, should it have its own unit tests? What is the common practice for the new-e2e?

A second suggestion, would it make sense to add an `Options` struct to it instead of passing the image as a string? That way we can extend to the necessary docker options if needed (`volume`, `network`, `command`, and so on)

Reply:
I don't think this would be considered `framework` functionality, but rather helpers for tests under `agent-platform`, we most definitely not have tests for these.

I think adding an `Options` struct here would be somewhat overkill given that it sounds unlikely that we'd ever add those here, given the scope of the tests (we're normally not supposed to install the Agent on a Docker container, that's what we release a dockerized Agent for). And if we did, this is currently only referenced once so making the change wouldn't be much effort.